lib/index.js

```diff
@@ -1,66 +1,50 @@
 'use strict';
 
-/**
- * Module dependencies
- */
 
-/* eslint-disable no-unused-vars */
-// Public node modules.
-const axios = require("axios");
-const FormData = require("form-data");
-const { Readable } = require('stream');
+
+const uploadVideo = require("./uploadVideo");
+const uploadImage = require("./uploadImage");
+const deleteFile = require("./deleteFile");
 
 module.exports = {
-  init(config) {
+  init: (config) => {
     const accountId = config.accountId;
     const apiKey = config.apiKey;
+    const variant = config.variant;
     return {
-      upload(file) {
-        const stream = new Readable({
-          read() {
-            this.push(file.buffer);
-            this.push(null);
-          },
-        })
-        var data = new FormData();
-        data.append("file", stream, `${file.hash}${file.ext}`);
-        return axios({
-          method: "POST",
-          url: `https://api.cloudflare.com/client/v4/accounts/${accountId}/images/v1`,
-          headers: {
-            Authorization: `Bearer ${apiKey}`,
-            ...data.getHeaders(),
-          },
-          data: data,
-        }).then((response) => {
-          const result = response.data.result;
-          const filename = result.filename;
-          const split = filename.split('.');
-          const type = split.length > 0 ? split[split.length - 1] : '';
-          file.url = result.variants[0];
-          file.provider_metadata = {
-            public_id: result.id,
-            resource_type: type,
-          };
-          return file;
-        });
+      upload: async (file) => {
+        switch (file.type.split('/')[0]) {
+          case 'video':
+            const uploadVideoResponse = await uploadVideo(file, accountId, apiKey);
+            const vresult = uploadVideoResponse;
```

Comment:
what is the point of this line?

Reply:
it might be better to extract the stream-media-id here as that is the only thing used below.

```
streamMediaId = uploadVideoResponse._response.headers['stream-media-id'];
```

Reply:
I think this was set before we decided what we were using it for, I've made it declare streamMediaId and use that below now, makes much more sense.